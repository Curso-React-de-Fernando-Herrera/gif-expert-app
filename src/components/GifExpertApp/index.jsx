import { useState } from "react"
import PropTypes from 'prop-types'

import AddCategory from 'components/AddCategory'
import GifGrid from "components/GifGrid"

const GifExpertApp = ({ initialCategory = ['Naruto'] }) => {
  const [categories, setCategories] = useState(initialCategory)

  return <div>
    <header>
      <h2>GifExpertApp</h2>
      <AddCategory
        setCategories={setCategories}
      />
      <hr />
    </header>
    <ol>
      {
        categories.map(category => (
          <GifGrid
            category={category}
            key={category}
          />
        ))
      }
    </ol>
  </div>
}

GifExpertApp.porpTypes = {
  initialCategory: PropTypes.arrayOf(PropTypes.string)
}

export default GifExpertApp
