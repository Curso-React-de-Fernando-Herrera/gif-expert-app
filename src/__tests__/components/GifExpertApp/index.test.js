import { shallow } from 'enzyme'
import GifExpertApp from 'components/GifExpertApp'

describe('<GifExpertApp /> component', () => {
  let wrapper = shallow(<GifExpertApp />)

  beforeEach(() => {
    wrapper = shallow(<GifExpertApp />)
  })

  test('should be rendered', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('should be show array of gifs', () => {
    const initialCategory = ['Sasuke', 'Itachi', 'Madara']

    wrapper = shallow(<GifExpertApp initialCategory={initialCategory} />)
    const returnedList = wrapper.find('GifGrid')

    expect( returnedList.length ).toBe( initialCategory.length )
    expect( returnedList.length ).not.toBe( initialCategory.length + 1 )
  })
})