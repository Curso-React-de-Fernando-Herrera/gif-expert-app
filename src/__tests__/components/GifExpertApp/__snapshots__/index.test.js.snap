// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifExpertApp /> component should be rendered 1`] = `
<div>
  <header>
    <h2>
      GifExpertApp
    </h2>
    <CreateCategory
      setCategories={[Function]}
    />
    <hr />
  </header>
  <ol>
    <GifGrid
      category="Naruto"
      key="Naruto"
    />
  </ol>
</div>
`;
